src/quickgui/plugin/qgsquickpositionmarker.qml

```diff
@@ -0,0 +1,172 @@
+/***************************************************************************
+  qgsquickpositionmarker.qml
+  --------------------------------------
+  Date                 : Dec 2017
+  Copyright            : (C) 2017 by Peter Petrik
+  Email                : zilolv at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick 2.3
+import QtQuick.Controls 2.2
+import QtQml 2.2
+import QtGraphicalEffects 1.0
+import QgsQuick 0.1 as QgsQuick
+
+/**
+ * \brief A marker refers to the current gps position according gps source is available. Along with longitude, latitude and radius.
+ */
+Item {
+  id: positionMarker
+  property int size: 48 * QgsQuick.Utils.dp
+
+  /**
+   * Note: required to be connected from parent!
+   */
+  property QgsQuick.MapSettings mapSettings
+  /**
+   * Utils for handling position.
+   */
+  property QgsQuick.PositionKit positionKit: QgsQuick.PositionKit {id: positionKit}
+  /**
+   * use in debug mode to simulate movement around some GPS location
+   * longitude, latitude, and radius, all in degrees WSG84
+   * e.g. simulatePositionLongLatRad = [60, 10, 0.02].
+   */
+  property var simulatePositionLongLatRad
+
+  /**
+   * Position in pixels
+   */
+  property point screenPosition
+  /**
+   * Color of the marker when gps is active.
+   */
+  property color baseColor: "darkblue"
+  /**
+   * Color of the marker when gps signal is lost.
+   */
+  property color unavailableColor: "gray"
+  property alias mapPosition: wgs84toMapCrs.projectedPosition // in map coordinates
+  property alias gpsPosition: positionKit.position // in WGS84 coordinates
+  property alias gpsAccuracy: positionKit.accuracy // in meters
+
+  /**
+   * Position info label.
+   */
+  property var gpsPositionLabel: qsTr(positionKit.gpsPositionLabel(3, "GPS signal lost"))
```

Comment:
I am thinking that all these could be just properties in the kit, we do not need aliases for all of those!